PowerShellTools/ServiceManagement/PowershellHostProcessHelper.cs

```diff
@@ -61,34 +106,94 @@ public static PowershellHostProcess CreatePowershellHostProcess()
+                    _powerShellHostProcess.Kill();
                 }
                 catch (Exception)
                 {
                 }
 
-                if (powershellHostProcess != null)
+                if (_powerShellHostProcess != null)
                 {
-                    powershellHostProcess.Dispose();
-                    powershellHostProcess = null;
+                    _powerShellHostProcess.Dispose();
+                    _powerShellHostProcess = null;
                 }
                 throw new PowershellHostProcessException(String.Format(CultureInfo.CurrentCulture,
                                                                         Resources.ErrorFailToCreateProcess,
                                                                         processId.ToString()));
             }
 
-            return new PowershellHostProcess(powershellHostProcess, endPointGuid);
+            _inputStreamWriter = _powerShellHostProcess.StandardInput;
+
+            return new PowerShellHostProcess(_powerShellHostProcess, EndPointGuid);
+        }
+
+        /// <summary>
+        /// Monitoring thread for user input request
+        /// </summary>
+        /// <remarks>
+        /// Will be started once app begins to run on remote PowerShell host service
+        /// Stopped once app exits
+        /// </remarks>
+        public static void MonitorUserInputRequest()
+        {
+            while (AppRunning)
+            {
+                foreach (ProcessThread thread in _powerShellHostProcess.Threads)
+                {
+                    if (thread.ThreadState == System.Diagnostics.ThreadState.Wait
+                        && thread.WaitReason == ThreadWaitReason.UserRequest)
+                    {
+                        if (PowerShellToolsPackage.Debugger != null &&
+                            PowerShellToolsPackage.Debugger.HostUi != null)
+                        {
+                            string inputText = PowerShellToolsPackage.Debugger.HostUi.ReadLine(Resources.UserInputRequestMessage);
+
+                            // Feed into stdin stream
+                            _inputStreamWriter.WriteLine(inputText);
+                            break;
+                        }
+                    }
+                }
+
+                Thread.Sleep(1000); 
+            }
+        }
+
+        private static void powerShellHostProcess_ErrorDataReceived(object sender, DataReceivedEventArgs e)
+        {
+            powerShellHostProcessOutput(e.Data);
+        }
+
+        private static void powerShellHostProcess_OutputDataReceived(object sender, DataReceivedEventArgs e)
+        {
+            powerShellHostProcessOutput(e.Data);
+        }
+
+        private static void powerShellHostProcessOutput(string outputData)
+        {
+            if (outputData.StartsWith(string.Format("[{0}]:", PowershellHostProcessHelper.EndPointGuid), StringComparison.OrdinalIgnoreCase))

```

Comment:
Do we want this StartsWith Formatting check/text to be in a Common location between this and when we log it?